stl/inc/__msvc_int128.hpp

```diff
@@ -0,0 +1,1155 @@
+// __msvc_int128 internal header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef __MSVC_INT128_HPP
+#define __MSVC_INT128_HPP
+
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#ifdef __cpp_lib_concepts
+#include <bit>
+#include <compare>
+#include <concepts>
+#include <cstdint>
+#include <intrin.h> // TODO: move things we need to intrin0
+
+#if 1 // FIXME: comment out _STL_INTERNAL_CHECKs and remove to make this core
+#include <yvals.h>
+#endif
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+struct _Int128;
+
+struct alignas(16) _Uint128 {
+    uint64_t _Word[2];
+
+    using _Signed_type   = _Int128;
+    using _Unsigned_type = _Uint128;
+
+    constexpr void _Left_shift(const unsigned char _Count) noexcept {
+        if (_Count >= 64) {
+            _Word[1] = _Word[0] << (_Count % 64);

```

Comment:
The behavior of integer-class types is specified in terms of a hypothetical extended integer type with the same width. I'm trying to avoid redundantly listing preconditions that are already present in the core language for integer types, e.g., `operator/` doesn't state that the divisor must be `0`. I didn't comment these because I felt the connection to the operator with its well-known preconditions was clear enough, but I don't mind doing so.

Reply:
Yeah, I figured that the same preconditions applied, but I thought that a comment would be helpful to indicate "yes, we knew about the preconditions when we wrote this code" instead of "we totally forgot that `_Count` might be large".